api-server/scanner-lib/src/blockchain_state/mod.rs

```diff
@@ -91,3 +128,162 @@ impl<S: ApiServerStorage + Send + Sync> LocalBlockchainState for BlockchainState
+                                                    .await?
+                                                    .unwrap_or(Amount::ZERO);
+
+                                                let new_amount = current_balance
+                                                    .sub(*amount)
+                                                    .ok_or_else(|| {
+                                                        ApiServerStorageError::LowLevelStorageError(
+                                                            "Balance should not underflow"
+                                                                .to_string(),
+                                                        )
+                                                    })?;
+
+                                                db_tx
+                                                    .set_address_balance_at_height(
+                                                        address.get(),
+                                                        new_amount,
+                                                        block_height,
+                                                    )
+                                                    .await?;
+                                            }
+                                        }
+                                    }
+                                    Destination::AnyoneCanSpend
+                                    | Destination::ClassicMultisig(_)
+                                    | Destination::ScriptHash(_) => {}
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    Ok(())
+}
+
+async fn update_balances_from_outputs<T: ApiServerStorageWrite>(
+    chain_config: Arc<ChainConfig>,
+    db_tx: &mut T,
+    block_height: BlockHeight,
+    outputs: &[TxOutput],
+) -> Result<(), ApiServerStorageError> {
+    for output in outputs {
+        match output {
+            TxOutput::Burn(_)
+            | TxOutput::CreateDelegationId(_, _)
+            | TxOutput::CreateStakePool(_, _)
+            | TxOutput::DataDeposit(_)
+            | TxOutput::DelegateStaking(_, _)
+            | TxOutput::IssueFungibleToken(_)
+            | TxOutput::IssueNft(_, _, _)
+            | TxOutput::ProduceBlockFromStake(_, _) => {}
+            TxOutput::Transfer(output_value, destination)
+            | TxOutput::LockThenTransfer(output_value, destination, _) => {
+                match destination {
+                    Destination::PublicKey(_) | Destination::Address(_) => {
+                        let address = Address::<Destination>::new(&chain_config, destination)
+                            .map_err(|_| {
+                                ApiServerStorageError::DeserializationError(
+                                    "Unable to encode destination".to_string(),
+                                )
```

Comment:
I think an expect() here is in order.

Reply:
I've converted all `map_err()` an `await?` in `mod.rs` to `.expect()`.

Reply:
I don't want to state the obvious, but obviously a blanket expect on everything isn't really what I meant. As long as you understand the circumstances and you find it OK, I'm fine with it.